src/client/options/mod.rs

```diff
@@ -673,41 +680,177 @@ impl Serialize for ClientOptions {
     }
 }
 
+/// Contains the options that can be set via a MongoDB connection string.
+///
+/// The format of a MongoDB connection string is described [here](https://docs.mongodb.com/manual/reference/connection-string/#connection-string-formats).
 #[derive(Debug, Default, PartialEq)]
-struct ClientOptionsParser {
-    pub hosts: Vec<ServerAddress>,
-    pub srv: bool,
+#[non_exhaustive]
+pub struct ConnectionString {
+    /// The initial list of seeds that the Client should connect to.
+    ///
+    /// Note that by default, the driver will autodiscover other nodes in the cluster. To connect
+    /// directly to a single server (rather than autodiscovering the rest of the cluster), set the
+    /// `direct_connection` field to `true`.
+    pub hosts: HostInfo,
```

Comment:
can we update the docstring to indicate that this can also contain a DNS name used for SRV lookups for the initial seed list? Also, I think renaming the field `host_info` might be more appropriate given that.

Reply:
Done.